setup-env.swift

```diff
@@ -0,0 +1,154 @@
+import Foundation
+
+// MARK: ConfiguratorError
+
+/// The possible errors that `ProjectConfigurator` can throw.
+public enum ConfiguratorError: LocalizedError {
+  /// The arguments provided to the configurator do not match the requirements
+  case invalidArguments
+
+  /// The source file providing the secret keys can not be found.
+  case keysFileNotFound
+
+  /// Could not read valid keys from the source file provided.
+  case emptyData
+
+  /// One of the keys in the source file was not found in the environment.
+  case keyMissing(String)
+
+  public var errorDescription: String? {
+    switch self {
+    case .invalidArguments:
+      return "The arguments provided to the configurator do not match the requirements"
+    case .keysFileNotFound:
+      return "The source file providing the secret keys can not be found."
+    case .emptyData:
+      return "Could not read valid keys from the source file provided."
+    case .keyMissing(let key):
+      return "The key \"\(key)\" is missing."
+    }
+  }
+}
+
+// MARK: EnvironmentConfigurator
+
+/// This class configures environment variables for the project.
+/// It takes a source file with the keys definition and inject the values
+/// to the provided output file.
+public final class EnvironmentConfigurator {
+
+  enum Defaults {
+    static let secretsFile = "secrets.xcconfig"
+    static let keysFile = "keys.env"
+  }
+
+  private let keysSource: String
+
+  private let fileManager = FileManager.default
+
+  public init(keysSource: String? = nil) {
+    self.keysSource = (keysSource ?? Defaults.keysFile).expandingTildeInPath
+  }
+
+  public func injectEnvironment(into file: String? = nil) throws {
+    let keys = try readKeys()
+
+    let fullPath = (file ?? Defaults.secretsFile).expandingTildeInPath
```

Comment:
```suggestion
  public func injectEnvironment(into file: String = Defaults.secretsFile) throws {
    let keys = try readKeys()

    let fullPath = file.expandingTildeInPath
```

Reply:
The `Defaults` enum is intended to be private, so can't be used as a parameter value in the signature. This approach is also convenient for allowing nil values from the command line